CONTRIBUTING.md

```diff
@@ -1,5 +1,128 @@
+   - Ensure yarn is installed
+   - Ensure your computer is setup for `React Native` (Will install the Android NDK if it's not present)
+   - Downloads the node modules (`yarn install`)
+
+3. Navigate to the demo React Native project `cd apps/bare-expo`
+4. Run the demo on any platform (Maybe start with web; it's the fastest! 😁)
+
+   - Web: `yarn web`
+   - iOS: `yarn ios`
+   - Android: `yarn android`
+
+> If this didn't work for you as described, please [open an issue.](https://github.com/expo/expo/issues/new/choose)
+
+## ✏️ Editing Packages
+
+All of our packages (Unimodules) can be found in the `packages/` directory. These packages are universally linked to the projects in the `apps/` directory (meaning any iOS, Android, web, or API changes can be tested from `apps/bare-expo/`). `bare-expo` is a Bare-Workflow React Native app that acts as a runner for the other projects in the `apps/` directory.
+
+- `native-component-list`: This is where you can write demos or tests that require physical interaction (A good playground for testing).
+- `test-suite`: You can write your E2E tests in here, when pushed to the remote, CI will run this project in Device Farm for Android, Detox for iOS, and Puppeteer for web!
+
+1. Navigate to a package you want to edit. Ex: `cd packages/expo-constants`
+2. Start the TypeScript build in watch mode: `yarn build`
+3. Edit code in that package's `src/` directory
+4. Play with your changes on a Simulator or device through `bare-expo`:
+   - Add or modify a file named after the API you're working on. Ex: `apps/test-suite/tests/Constants.js`
+   - Run the code from the `bare-expo` project with `yarn <android | ios | web>` or test the code with `yarn test:<ios | web>`
+5. You can edit the native code by opening the projects in their respective editors:
+   - Android Studio: `yarn edit:android`
+   - Xcode: `yarn edit:ios`
+   - Remember to **rebuild** the native project whenever you make a native change
+
+**Extra Credit**
+
+- [Guide to Unimodule Development](guides/Expo%20Universal%20Module%20Infrastructure.md)
+- [Expo JS Style Guide](guides/Expo%20JavaScript%20Style%20Guide.md) (also mostly applies to TypeScript)
+- The React Native dev tools are currently disabled in our fork [#5602](https://github.com/expo/expo/issues/5602). You can hack around this by cloning React Native outside this repo, then copying the contents `react-native/React/DevSupport` into `expo/react-native-lab/react-native/React/DevSupport` (This will only enable the shake gesture, CMD+R won't work yet).
+- We use a fork of `react-native` in this repo, this fork is located at `react-native-lab/react-native` (you can make changes or cherry-picks from here if you want).
+- All of the package's `build/` code should be committed. This is because working with native code can make things very unpredictable, we found this is the best way to cut down on confusion.
```

Comment:
```suggestion
- All of the package's `build/` code should be committed. This is because it is simpler to reproduce issues if all contributors are running the same code and so we don't need to rebuild dozen of packages locally on every `git pull` or `git checkout` operation.
```